nibabel/cifti2/cifti2.py

```diff
@@ -0,0 +1,848 @@
+            elif len(metadata) == 2:
+                pairs = [tuple((metadata[0], metadata[1]))]
+            else:
+                raise ValueError('nvpair must be a 2-list or 2-tuple')
+        else:
+            raise ValueError('nvpair input must be a list, tuple or dict')
+
+        for pair in pairs:
+            if func == 'add':
+                if pair not in self.data:
+                    self.data.append(pair)
+            elif func == 'remove':
+                self.data.remove(pair)
+            else:
+                raise ValueError('Unknown func %s' % func)
+
+    def add_metadata(self, metadata):
+        """Add metadata key-value pairs
+
+        This allows storing multiple keys with the same name but different
+        values.
+
+
+        Parameters
+        ----------
+        metadata : 2-List, 2-Tuple, Dictionary, List[2-List or 2-Tuple]
+                     Tuple[2-List or 2-Tuple]
+
+        Returns
+        -------
+        None
+
+        """
+        if metadata is None:
+            return
+        self._add_remove_metadata(metadata, 'add')
+
+    def remove_metadata(self, metadata):
+        if metadata is None:
+            return
+        self._add_remove_metadata(metadata, 'remove')
+
+    def _to_xml_element(self):
+        metadata = xml.Element('MetaData')
+
+        for name_text, value_text in self.data:
+            md = xml.SubElement(metadata, 'MD')
+            name = xml.SubElement(md, 'Name')
+            name.text = str(name_text)
+            value = xml.SubElement(md, 'Value')
+            assert(value_text is not None)
+            value.text = str(value_text)
+        return metadata
+
+
+class Cifti2LabelTable(xml.XmlSerializable):
+
+    def __init__(self):
+        self.labels = []
+
+    @property
+    def num_labels(self):
+        return len(self.labels)
+
+    def get_labels_as_dict(self):
+        self.labels_as_dict = {}
+        for ele in self.labels:
+            self.labels_as_dict[ele.key] = ele.label
+        return self.labels_as_dict
+
+    def _to_xml_element(self):
+        assert len(self.labels) > 0
+        labeltable = xml.Element('LabelTable')
+        for ele in self.labels:
+            labeltable.append(ele._to_xml_element())
+        return labeltable
+
+    def print_summary(self):
+        print(self.get_labels_as_dict())
+
+
+class Cifti2Label(xml.XmlSerializable):
+
+    def __init__(self, key=0, label='', red=None,
+                 green=None, blue=None, alpha=None):
+        self.key = key
+        self.label = label
+        self.red = red
+        self.green = green
+        self.blue = blue
+        self.alpha = alpha
+
+    @property
+    def rgba(self):
+        """ Returns RGBA as tuple """
+        return (self.red, self.green, self.blue, self.alpha)
+
+    def _to_xml_element(self):
+        lab = xml.Element('Label')
+        lab.attrib['Key'] = str(self.key)
+        lab.text = str(self.label)
+        if self.red is not None:
```

Comment:
Can any of these values be none, in fact?   E.g red = None, green = None, blue = 1, alpha = 1?